### PR TITLE
rspamd: fix builds on non-x86_64-linux platforms

### LINKED ISSUE #102751: rspamd for aarch64 is broken on latest unstable
**Describe the bug**
rspamd for aarch64 can't find luajit and fails to build.

**To Reproduce**
`nix build github:NixOS/nixpkgs#legacyPackages.aarch64-linux.rspamd`

At the moment of writing, the latest unstable seems to be `34ad166a830d3ac1541dcce571c52231f2f0865a`.

**Expected behavior**
Build succeeds

**Screenshots**
n/a

**Additional context**
Hydra build log: https://hydra.nixos.org/build/128842971/nixlog/1

**Notify maintainers**
@avnik @fpletz @globin

**Metadata**
Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
 - rspamd
# a list of nixos modules affected by the problem
module:
 - rspamd
```

### PR DESCRIPTION
**Pull request by vikanezrimaya**
###### Motivation for this change
Fixes #102751 - architectures other than `x86_64` need some love too! (also, it's a package I use on my production machine and I finally figured out how to fix it)

LuaJIT is built in rspamd only on x86_64-linux, and LuaJIT support became enabled by default in 2.6, breaking builds without it. This commit explicitly disables LuaJIT support on non-x86_64 architectures.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS (i686-linux, x86_64-linux, aarch64-linux)
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - `nixpkgs#legacyPackages.x86_64-linux.rspamd.tests.rspamd.simple`
  - `nixpkgs#legacyPackages.i686-linux.rspamd.tests.rspamd.simple`
  - `nixpkgs#legacyPackages.aarch64-linux.rspamd.tests.rspamd.simple`
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @avnik @fpletz @globin

### CODE REVIEW

**Review by ajs124 — APPROVED** [review]

**Comment by kisik21** [comment]
@ajs124 this PR uses an experimental bot to accelerate code review. If you would merge this PR if you had permissions, write `/status needs_merger` in a comment to accelerate the process per marvin-bot's usage manual: https://github.com/timokau/marvin-mk2/blob/deployed/USAGE.md :sparkles:

**Review by nh2 — APPROVED** [review]
Change looks good, but the ofborg is still building, and might not finish because it seems to be building the world on aarch64:

https://logs.nix.ci/?attempt_id=f70c6e0c-25ad-4986-b237-685aa8e6ac5f&key=nixos%2Fnixpkgs.103973

@kisik21 What do I need to tell the Marvin if I do have merge permissions and would merge, but first want to give the ofborg the chance to finish the build (e.g. "merge if CI goes green")?

**Comment by kisik21** [comment]
@timokau consider giving Marvin merge perms and an auto-merge command on ofborg approval?

**Comment by kisik21** [comment]
Also you could try to build this PR yourself - it's a fairly small change, and if you're on unstable it should build pretty quick :3

**Comment by timokau** [comment]
@kisik21 no, I consider that out of scope. Its not necessary for marvin's core functionality ("Making sure your PR gets a review and your reviews don't get lost.") and I don't want to have that kind of responsibility.

A merge bot has been / is discussed a lot though, for example [this](https://discourse.nixos.org/t/steam-engine-bors-for-merge-train/9676) might interest you.

@nh2 it doesn't have any support for that right now. I could imagine a simply reminder functionality at some point in the future. I have thought about some ofBorg integration anyway (it could ask the PR author to fix ofBorg errors before requesting a reviewer). Reminders are pretty far down the backlog for now though.

**Comment by timokau** [comment]
Since it was broken before and I assume @kisik21 has tested this locally, I'll ignore the running ofBorg check now. It has been queued for 6h already.

Thanks for fixing this!

**Review by timokau — APPROVED** [review]

**Comment by nh2** [comment]
> it doesn't have any support for that right now

@timokau @kisik21 My request isn't to allow the bot to auto-merge (that'd be nice for another day), but to address the fact that the [usage](https://github.com/timokau/marvin-mk2/blob/deployed/USAGE.md) doesn't describe what to do. It says:

> if the PR has been reviewed and the reviewer would merge this PR if they could, but does not have the permission to do so

so the first thing any commiter is thinking is _"'does not have the permission to do so'_ doesn't apply to me, so what do I do now, well the document doesn't say, so I guess ... nothing?"

If "nothing" is the right answer, it'd still be nice to write that into the doc.

**Comment by timokau** [comment]
Marvin just augments the usual PR review process. So in cases where it can't help (yet), the usual process (or lack thereoff) applies. In my case that would be "keep a tab open for a while, then check if something might be wrong with ofBorg if the check still hasn't passed".

Still, any contribution to or clarification of the contributing document is appreciated, feel free to [open](https://github.com/timokau/marvin-mk2) a PR :)

**Pull request merged**
